### ISSUE TITLE
Realtime Collaboration Error

### ISSUE THREAD

**Issue by EpicJosch** [jgraph/drawio-nextcloud#52]
When i create a new Diagram it opens up and then i get this RealTime Collaboration Error:

![grafik](https://github.com/jgraph/drawio-nextcloud/assets/50040609/de9b34a6-5ac4-4bd4-80c3-09fae0200f61)

With this error i cant add objects to the diagram so i cant edit it.

I have to close and reopen the file then it works.

Is this a known error?

**Comment by davidjgraph** [comment]
versions of drawio and nextcloud? browser and version? errors in console?

**Comment by EpicJosch** [comment]
Nextcloud Version: 26.0.3 (Newest Stable) (But error also occures on Version 25.*)
Drawio Version: 2.1.2 (Newest)
Browser: Chrome (but error also occures on Firefox and Edge)
Browser Version: 114.0.5735.134 (Official Build) (64-Bit)
Errors in Console: There are no errors in console

**Comment by m-mohamedin** [comment]
Do you use the default draw.io URL? Do you network settings (e.g, firewall) blocking web sockets?
Can you please open the Network tab in the browser console and send us any request that failed.
Thanks

**Comment by EpicJosch** [comment]
Yeah i use the default draw.io URL. The only 2 errors i get in the Browser Developer Tab are those:

![errors](https://github.com/jgraph/drawio-nextcloud/assets/50040609/7e95dcf7-bd13-4506-9f17-d7542554ddd6)

**Comment by mariusbolik** [comment]
Having the same issue. I am on NC 26.0.3 and using Drawio 2.1.2.
The app isn't working for me. No adblock enabled.

**Comment by piiskop** [comment]
I concur that the issue exists. Nothing can be done. Not a single element can be dragged and dropped.

**Comment by arnowelzel** [comment]
> Yeah i use the default draw.io URL. The only 2 errors i get in the Browser Developer Tab are those:
> 
> ![errors](https://user-images.githubusercontent.com/50040609/255313920-7e95dcf7-bd13-4506-9f17-d7542554ddd6.png)
> 
> Edit: Also i don't think that my network settings block web sockets since they run on port 80 and 443 which are opened in my firewall

See the message: `net::ERR_BLOCKED_BY_CLIENT`. This is a clear indication that your client blocks the requests. Maybe you have an ad blocker or some security addon or setting which blocks that.

Also this this discussion: https://stackoverflow.com/questions/23341765/getting-neterr-blocked-by-client-error-on-some-ajax-calls

**Comment by piiskop** [comment]
I do not have an adblocker other than the one for YouTube.
If I reload the page then clicking shapes and bringing them to the canvas works. Without reload, nothing works. I do not see the given blocking error message.

**Comment by davidjgraph** [comment]
You don't need log.diagrams.net, that really is just logging. Check whether web socket connections are allowed to app.diagrams.net.

**Comment by andrewlixin1** [comment]
I have encountered the following error：

[drawio] 错误: OCP\Files\NotFoundException:  at <<closure>>
0. /var/www/nextcloud/apps/drawio/lib/Controller/EditorController.php line 202
   OCA\Drawio\Controller\EditorController->getFileById()
1. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 225
   OCA\Drawio\Controller\EditorController->getFileRevisions()
2. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 133
   OC\AppFramework\Http\Dispatcher->executeController()
3. /var/www/nextcloud/lib/private/AppFramework/App.php line 172
   OC\AppFramework\Http\Dispatcher->dispatch()
4. /var/www/nextcloud/lib/private/Route/Router.php line 298
   OC\AppFramework\App::main()
5. /var/www/nextcloud/lib/base.php line 1047
   OC\Route\Router->match()
6. /var/www/nextcloud/index.php line 36
   OC::handleRequest()

GET /nextcloud/index.php/apps/drawio/ajax/getFileRevisions?fileId=%7BfileId%7D

**Comment by arnowelzel** [comment]
That looks like the file got deleted while you are still working on it.

**Comment by EpicJosch** [comment]
> 

yeah but thats not the case. I create the file and the editor opens. Then the issue appears and after i close and reopen the file it works..

**Comment by arnowelzel** [comment]
I think the reason for this error is connected to https://github.com/jgraph/drawio-nextcloud/issues/57. When creating a new diagram it can not be edited before reloading it at least once.
